core/lib/dal/src/storage_web3_dal.rs

```diff
@@ -69,17 +69,37 @@ impl StorageWeb3Dal<'_, '_> {
 
     /// Gets the current value for the specified `key`.
     pub async fn get_value(&mut self, key: &StorageKey) -> DalResult<H256> {
-        self.get_historical_value_unchecked(key, L2BlockNumber(u32::MAX))
+        let Some(l2_block_number) = self
+            .storage
+            .blocks_dal()
+            .get_sealed_l2_block_number()
+            .await?
+        else {
+            return Ok(H256::zero());
```

Comment:
This doesn't work with snapshot recovery, does it? Ditto for `get_values` below. I think this is acceptable because `*web3_dal` methods should be called once there's at least one local L1 batch, but still IMO this should be at least documented, and returning an error doesn't seem to have downsides. (Or you would use something like resolving `BlockNumber::Pending` and subtracting one, possibly at the call sites, so that these methods do work with snapshot recovery.)

Reply:
Agree, returning error makes sense